java-frontend/src/main/java/org/sonar/java/se/checks/ExceptionalYieldChecker.java

```diff
@@ -98,18 +101,37 @@ private void reportIssue(ExplodedGraph.Node node, ExceptionalCheckBasedYield yie
     check.reportIssue(reportTree, String.format(message, methodName), flows.build());
   }
 
-  private static Set<List<JavaFileScannerContext.Location>> flowsForMethodArguments(ExplodedGraph.Node node, MethodInvocationTree mit) {
+  private static Set<List<JavaFileScannerContext.Location>> flowsForMethodArguments(ExplodedGraph.Node node, MethodInvocationTree mit, int parameterCausingExceptionIndex) {
     ProgramState programState = node.programState;
     List<ProgramState.SymbolicValueSymbol> arguments = Lists.reverse(programState.peekValuesAndSymbols(mit.arguments().size()));
-    List<Class<? extends Constraint>> domains = domainsFromArguments(programState, arguments);
-    Set<SymbolicValue> argSymbolicValues = arguments.stream().map(ProgramState.SymbolicValueSymbol::symbolicValue).collect(Collectors.toCollection(LinkedHashSet::new));
-    Set<Symbol> argSymbols = arguments.stream().map(ProgramState.SymbolicValueSymbol::symbol).filter(Objects::nonNull).collect(Collectors.toCollection(LinkedHashSet::new));
+    SymbolicValue parameterCausingExceptionSV = arguments.get(parameterCausingExceptionIndex).symbolicValue();
+
+    Set<SymbolicValue> argSymbolicValues = new LinkedHashSet<>();
+    Set<Symbol> argSymbols = new LinkedHashSet<>();
+    for (ProgramState.SymbolicValueSymbol svs : arguments) {
+      Symbol symbol = svs.symbol();
+      SymbolicValue sv = svs.symbolicValue();
+      if (parameterCausingExceptionSV == sv) {
+        argSymbolicValues.add(sv);
+      } else {
+        PMap<Class<? extends Constraint>, Constraint> constraints = programState.getConstraints(sv);
+        if (constraints == null || (domainsFromConstraints(constraints).size() == 1 && constraints.get(ObjectConstraint.class) == ObjectConstraint.NOT_NULL)) {
+          // skip SV without constraints or only having a single NOT_NULL constraint
+          continue;
+        }
+        argSymbolicValues.add(sv);
+      }
+      if (symbol != null) {
```

Comment:
Symbols should be tracked iff symbolic value is being tracked, this will cause tracking of symbols we are not interested in. I am bit surprised that this wasn't in the tests already, add test if this is missing.